src/modules/workspace/promote.ts

```diff
@@ -1,45 +1,74 @@
-import chalk from 'chalk'
+import axios from 'axios'
+import boxen from 'boxen'
+import ora from 'ora'
 import { createFlowIssueError } from '../../api/error/utils'
 import { createWorkspacesClient } from '../../api/clients/IOClients/infra/Workspaces'
 import { SessionManager } from '../../api/session/SessionManager'
 import log from '../../api/logger'
 import { promptConfirm } from '../../api/modules/prompts'
+import { VBase } from '../../api/clients/IOClients/infra/VBase'
+import authUrl from '../url'
 import useCmd from './use'
+import { COLORS } from '../../lib/constants/Colors'
+import { Messages } from '../../lib/constants/Messages'
 
+const { checkForConflicts } = VBase.createClient()
 const { promote, get } = createWorkspacesClient()
 const { account, workspace: currentWorkspace } = SessionManager.getSingleton()
+const workspaceUrl = authUrl()
 
 const throwIfIsMaster = (workspace: string) => {
   if (workspace === 'master') {
-    throw createFlowIssueError(`It is not possible to promote workspace ${workspace} to master`)
+    throw createFlowIssueError(Messages.PROMOTE_MASTER_ERROR(workspace))
+  }
+}
+
+const handleConflict = async () => {
+  const conflictsFound = await checkForConflicts()
+  if (conflictsFound) {
+    await axios.get(workspaceUrl)
   }
 }
 
 const isPromotable = async (workspace: string) => {
+  const spinner = ora(Messages.PROMOTE_SPINNER_START).start()
+  spinner.color = COLORS.MAGENTA
+
   throwIfIsMaster(workspace)
+
   const meta = await get(account, currentWorkspace)
   if (!meta.production) {
-    throw createFlowIssueError(
-      `Workspace ${chalk.green(currentWorkspace)} is not a ${chalk.green(
-        'production'
-      )} workspace\nOnly production workspaces may be promoted\nUse the command ${chalk.blue(
-        'vtex workspace create <workspace> --production'
-      )} to create a production workspace`
-    )
+    throw createFlowIssueError(Messages.PROMOTE_NOT_PRODUCTION_ERROR(workspace))
   }
+
+  await handleConflict()
+
+  spinner.succeed()
 }
+
 const promptPromoteConfirm = (workspace: string): Promise<boolean> =>
-  promptConfirm(`Are you sure you want to promote workspace ${chalk.green(workspace)} to master?`, true)
+  promptConfirm(Messages.PROMOTE_PROMPT_CONFIRM(workspace), true)
 
 export default async () => {
-  log.debug('Promoting workspace', currentWorkspace)
+  log.debug(Messages.PROMOTE_INIT, currentWorkspace)
+
   await isPromotable(currentWorkspace)
+  console.log(Messages.PROMOTE_CHECK_WORKSPACE(currentWorkspace, workspaceUrl))
```

Comment:
Why some log.info and some console.log?

Reply:
Might get it, you don't want the info: tag.

Reply:
Yes, also we don't need this info in the debug file